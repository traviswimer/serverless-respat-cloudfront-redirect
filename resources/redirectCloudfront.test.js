const redirectCloudfront = require('./redirectCloudfront');

describe('redirectCloudfront()', () => {
	test('generates distribution', () => {
		expect(redirectCloudfront({
			prefix: "ThePrefix",
			pattern_name: "ThePatternName",
			redirect_from: "TheRedirectFrom",
			region: "TheRegion",
			acm_certificate_arn: "TheCertArn"
		})).toMatchSnapshot();
	});
});
