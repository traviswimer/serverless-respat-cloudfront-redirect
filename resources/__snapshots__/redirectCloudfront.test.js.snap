// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redirectCloudfront() generates distribution 1`] = `
Object {
  "DependsOn": Array [
    "ThePatternNameRedirectBucket",
  ],
  "Properties": Object {
    "DistributionConfig": Object {
      "Aliases": Array [
        "TheRedirectFrom",
      ],
      "Comment": "Cloudfront Distribution pointing to 'TheRedirectFrom' S3 redirect bucket",
      "DefaultCacheBehavior": Object {
        "AllowedMethods": Array [
          "GET",
          "HEAD",
        ],
        "Compress": true,
        "ForwardedValues": Object {
          "Cookies": Object {
            "Forward": "all",
          },
          "QueryString": true,
        },
        "TargetOriginId": "ThePatternNameCloudfrontRedirectBucketOrigin",
        "ViewerProtocolPolicy": "redirect-to-https",
      },
      "Enabled": true,
      "HttpVersion": "http2",
      "IPV6Enabled": true,
      "Origins": Array [
        Object {
          "CustomOriginConfig": Object {
            "HTTPPort": "80",
            "HTTPSPort": "443",
            "OriginProtocolPolicy": "http-only",
          },
          "DomainName": Object {
            "Fn::Join": Array [
              "",
              Array [
                "TheRedirectFrom",
                ".s3-website-TheRegion.amazonaws.com",
              ],
            ],
          },
          "Id": "ThePatternNameCloudfrontRedirectBucketOrigin",
        },
      ],
      "PriceClass": "PriceClass_100",
      "ViewerCertificate": Object {
        "AcmCertificateArn": "TheCertArn",
        "SslSupportMethod": "sni-only",
      },
    },
  },
  "Type": "AWS::CloudFront::Distribution",
}
`;
